include/swift/SIL/SILInstruction.h

```diff
@@ -1222,12 +1222,14 @@ class StringLiteralInst final : public LiteralInst,
 private:
   unsigned Length;
   Encoding TheEncoding;
+  bool NullTerminated;
 
   StringLiteralInst(SILDebugLocation *DebugLoc, StringRef text,
-                    Encoding encoding, SILType ty);
+                    Encoding encoding, bool nullTerminate, SILType ty);
```

Comment:
That approach breaks a few tests which require literals to be null terminated, but which assert-fail in llvm if a null is actually included in the StringRef.

Getting rid of this flag makes SLI->getValue() more complicated, and forces the frontend to make copies of string literals to ensure that they are null-terminated.